detox/src/ios/expectTwo.test.js

```diff
@@ -453,13 +459,47 @@ describe('expectTwo', () => {
     expect(testCall).deepEquals(jsonOutput);
   });
 
-  it(`element().takeScreenshot`, () => {
-    try {
-      e.element(e.by.id('uniqueId')).takeScreenshot();
-      fail();
-    } catch (e) {
-      expect(e.message).toContain('not supported on iOS');
-    }
+  it(`element().takeScreenshot check for path`, async () => {
+    const tmpFilePath = 'pathToSomeFile.jpg'
+    require('tempfile').mockReturnValue(tmpFilePath);
+    const testCall = await e.element(e.by.id('uniqueId')).takeScreenshot('imageName');
+
+    expect(testCall).toBe(tmpFilePath);
+  });
+
+  it(`element().takeScreenshot without a name`, async () => {
+    const tmpFilePath = 'pathToSomeFile.jpg'
+    require('tempfile').mockReturnValue(tmpFilePath);
+    await e.element(e.by.id('uniqueId')).takeScreenshot();
+
+    const jsonOutput = {
+      type: 'action',
+      action: 'takeScreenshot',
+      predicate: {
+        type: 'id',
+        value: 'uniqueId'
+      }
+    };
+
+    expect(invocationManager.execute).toHaveBeenCalledWith(jsonOutput);
+  });
+
+  it(`element().takeScreenshot with name`, async () => {
+    await e.element(e.by.id('uniqueId')).takeScreenshot('imageName');
+
+    const jsonOutput = {
+      type: 'action',
+      action: 'takeScreenshot',
+      params: [
+        'imageName'
+      ],
+      predicate: {
+        type: 'id',
+        value: 'uniqueId'
+      }
+    };
+
+    expect(invocationManager.execute).toHaveBeenCalledWith(jsonOutput);
   });
```

Comment:
@alon-ha could you please add tests so as to see that the emitter call is properly done, and that indeed the file is moved to the right path?

Reply:
I don't think it's related to here. Such tests should be related to the emitter and the artifacts functionality

Reply:
Implemented.